controllers/tinkerbellcluster_controller.go

```diff
@@ -49,7 +49,7 @@ type TinkerbellClusterReconciler struct {
 
 // validate validates if context configuration has all required fields properly populated.
 func (tcr *TinkerbellClusterReconciler) validate() error {
-	if tcr.Client == nil {
+	if tcr == nil {
```

Comment:
This kind of check is largely irrelevant and is an example of a programmer misusing the construct. I'd vote for removing the check.